lib/fixed-points.nix

```
@@ -92,15 +92,22 @@ rec {
     0
     ```
 
+    :::{.warning}
+    This function is deprecated, see <https://github.com/NixOS/nixpkgs/pull/275760> for alternatives.
```

```suggestion
    This function is **inefficient**, and must only be used when no better algorithm exists for the problem at hand.
    The inefficiency stems from
    - performing potentially very large equality checks on each iteration, leading to O(n²) complexity,
    - equality checks may defeat laziness in the returned data structure,
    - and the inability to carry over already computed information from one iteration to the next.
    Instead, consider using
    - `fix`, if `f` is not strict in its argument
```

Also add the suggestions from the PR description.